contrib/lkt/language/parser.py

```diff
@@ -141,14 +141,14 @@ def root_get(entity_name=T.Symbol):
 
     @langkit_property(return_type=T.TypeDecl.entity)
     def get_builtin_type(entity_name=T.Symbol):
-        return Self.root_get(entity_name).cast(T.TypeDecl)
+        return Self.root_get(entity_name).cast_or_raise(T.TypeDecl)
```

Comment:
I agree with @pmderodat that this would go better in another commit

As for the testcase, not sure it's feasible nor useful: The reason we're using `cast_or_raise` here is because this is used in internal methods where we expect to return a type. Here, it made the `node_type` property return `None`, which is clearly not great, rather than raising an exception, which is better IMHO because it announces that underlying assumptions have been broken.

Since it's an internal property it's not easy to test.

Reply:
OK. I'll move them in another commit then! For the test, let me know if I should add one!

Reply:
Fair enough, I agree with Raphaël. Thanks! I suggest adding a justification to the commit message for such cases (here, what Raph said is good).

Shouldn’t we do `cast_or_raise`in `root_get` as well, by the way?

Reply:
Done. Also added `cast_or_raise` in `root_get`.